### PR TITLE
feat: fix pandas types in docstrings/python types

### PR DESCRIPTION
**Pull request by Mause**

### CODE REVIEW

File: tools/pythonpkg/duckdb-stubs/__init__.pyi [hunk 1]
```diff
@@ -37,32 +37,30 @@ class DataError(Error): ...
...
     def close(self) -> None: ...
     def commit(self) -> DuckDBPyConnection: ...
     def cursor(self) -> DuckDBPyConnection: ...
     @overload
-    def df(self) -> object: ...
+    def df(self) -> pandas.DataFrame: ...
     @overload
-    def df(self, df: object) -> DuckDBPyRelation: ...
-    @overload
-    def df(aliasoffrom_df) -> Any: ...
```
Comment (Mause) [inline tools/pythonpkg/duckdb-stubs/__init__.pyi:51]: This function has been removed now, as it never actually existed - the mypy stub generator was confused by the braces in the docstring

Comment (pdet) [reply tools/pythonpkg/duckdb-stubs/__init__.pyi:51]: (Sligthly unrelated to the PR, but triggered by this comment) Looking at these functions that are a lot of operators (e.g., limit) specialized for DFs, I guess it comes from a time before we had a RAPI, I wonder if it makes sense to keep these. I guess ideally users would need to transform dfs to duckdb rel, use our internal operators, and then output the result to whatever format they want.

**Comment by pdet** [comment]
In general, lgtm.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**